playwright_tests/E2E/test-case.test.ts

```diff
@@ -0,0 +1,113 @@
+import { test, expect } from '@playwright/test';
+import { signIn, signOut } from './steps/login-steps';
+import { waitForSpinner } from './steps/spinner-steps';
+import { getCaseReferenceFromFirstRow } from './steps/table-steps';
+import { confirmNextSteps, confirmTabsVisible, caseDetailsCheck, validateWorkBasketComplexValues, validateWorkbasketInputs } from './steps/test-case';
+import { waitForSpecificResponse } from './helpers/responseListenerHelper';
+
+test('Validate next steps drop down', async ({ page }) => {
+  const response = waitForSpecificResponse(
+    page,
+    'data/internal/cases/',
+    'GET'
+  );
+
+  await signIn(page, 'SOLICITOR');
+  await expect(page.getByLabel('Manage Cases')).toBeVisible();
+  await page.getByLabel('Case type').selectOption({ label: 'XUI Case PoC' });
+  await page.getByLabel('Apply filter').click();
+  await waitForSpinner(page);
+  await expect(page.getByRole('heading', { name: 'Your cases' })).toBeVisible();
+  const firstCaseRef = await getCaseReferenceFromFirstRow(page);
+  await page.getByLabel(`go to case with Case reference:${firstCaseRef.trim(4)}`).click();
+  const responseData = await response;
+  const expectedData = responseData?.triggers;
+  const nextStepsMatch = await confirmNextSteps(page, expectedData);
+  expect(nextStepsMatch).toBeTruthy();
+  await signOut(page);
+});
+
+test('Validate tabs are visible', async ({ page }) => {
+  const response = waitForSpecificResponse(
+    page,
+    'data/internal/cases/',
+    'GET'
+  );
+
+  await signIn(page, 'SOLICITOR');
+  await expect(page.getByLabel('Manage Cases')).toBeVisible();
+  await page.getByLabel('Case type').selectOption({ label: 'XUI Case PoC' });
+  await page.getByLabel('Apply filter').click();
+  await waitForSpinner(page);
+  await expect(page.getByRole('heading', { name: 'Your cases' })).toBeVisible();
+  const firstCaseRef = await getCaseReferenceFromFirstRow(page);
+  await page.getByLabel(`go to case with Case reference:${firstCaseRef.trim(4)}`).click();
```

Comment:
What does trim(4) do here?

Reply:
Updated code to use function to remove 4, this happens as some labels have the last 4 digit of case ref removed